bootloader/src/pyi_utils.c

```diff
@@ -850,21 +855,44 @@ set_systemd_env()
 pid_t child_pid = 0;
 
 static void
-_signal_handler(int signal)
+_ignore_signal(int signum)
+{
+    VS("LOADER: Ignoring signal %d\n", signum);
+}
+
+static void
+_signal_handler(int signum)
 {
-    kill(child_pid, signal);
+    VS("LOADER: Forwarding signal %d to child pid %d\n", signum, child_pid);
+    kill(child_pid, signum);
 }
 
 /* Start frozen application in a subprocess. The frozen application runs
  * in a subprocess.
  */
 int
-pyi_utils_create_child(const char *thisfile, const int argc, char *const argv[])
+pyi_utils_create_child(const char *thisfile, const ARCHIVE_STATUS* status,
+                       const int argc, char *const argv[])
 {
     pid_t pid = 0;
     int rc = 0;
     int i;
 
+    /* As indicated in signal(7), signal numbers range from 1-31 (standard)
+     * and 32-64 (Linux real-time). */
+    const size_t num_signals = 65;
+
+    #ifndef sighandler_t
+    typedef void (*sighandler_t)(int);
+    #endif
+    sighandler_t handlers[num_signals];
+    sighandler_t handler;
+    int ignore_signals;
+    int signum;
+
+    ignore_signals = (pyi_arch_get_option(status, "pyi-bootloader-ignore-signals") != NULL);
+    handler = ignore_signals ? &_ignore_signal : &_signal_handler;
```

Comment:
I suggest moving this to the parent code, since it doe not affect the child's behaviour.